project01/dominik/game.cc

```diff
@@ -0,0 +1,65 @@
+#include "game.h"
+
+static const int REWARD_FOR_BAIT = 10;
+
+Game::Game(const cinder::ivec2& screenSize, const int snakeBodyPartSize)
+    : mFieldSize(screenSize),
+      mSnake(mFieldSize, snakeBodyPartSize),
+      mBait(mFieldSize),
+      mState(GameState::running),
+      mScore(0)
+{}
+
+void Game::Update()
+{
+    mSnake.Move();
+
+    if (mSnake.HeadClashedWithBody() || mSnake.HeadOutsideField())
+    {
+        std::cout << "Head clashed with body or outside field!" << std::endl;
+        mState = GameState::lost;
+        return;
+    }
+    if (mSnake.MinDistanceHeadToVectorNowAndBefore(mBait.GetPosition()) < static_cast<float>(mBait.GetRadius() + mSnake.GetSize()))
+    {
+        mScore += REWARD_FOR_BAIT;
+        std::cout << "Snake ate bait. New score: " << mScore << std::endl;
+        mSnake.Grow();
+
+        if(mScore % (REWARD_FOR_BAIT * 2) == 0)
```

Comment:
Nice. Is this part of the "original" snake game? I forgot to implement this in mine.